lib/core.ps1

```diff
@@ -42,6 +50,76 @@ function Get-UserAgent() {
     return "Scoop/1.0 (+http://scoop.sh/) PowerShell/$($PSVersionTable.PSVersion.Major).$($PSVersionTable.PSVersion.Minor) (Windows NT $([System.Environment]::OSVersion.Version.Major).$([System.Environment]::OSVersion.Version.Minor); $(if($env:PROCESSOR_ARCHITECTURE -eq 'AMD64'){'Win64; x64; '})$(if($env:PROCESSOR_ARCHITEW6432 -eq 'AMD64'){'WOW64; '})$PSEdition)"
 }
 
+function load_cfg($file) {
+    if(!(Test-Path $file)) {
+        return $null
+    }
+
+    try {
+        return (Get-Content $file -Raw | ConvertFrom-Json -ErrorAction Stop)
+    } catch {
+        Write-Host "ERROR loading $file`: $($_.exception.message)"
+    }
+}
+
+$scoopConfig = load_cfg $configFile
+
+function get_config($name, $default) {
+    if($null -eq $scoopConfig.$name -and $null -ne $default) {
+        return $default
+    }
+    return $scoopConfig.$name
+}
+
+function set_config($name, $val) {
+    if(!$scoopConfig) {
+        $scoopConfig = @{ $name = $val }
+    } else {
+        if($val -eq [bool]::TrueString -or $val -eq [bool]::FalseString) {
+            $val = [System.Convert]::ToBoolean($val)
+        }
+        $scoopConfig.$name = $val
+    }
+
+    if($null -eq $val) {
+        $scoopConfig.PSObject.Properties.Remove($name)
+    }
+
+    ConvertTo-Json $scoopConfig | Set-Content $configFile -Encoding utf8
```

Comment:
```suggestion
    ConvertTo-Json $scoopConfig | Set-Content $configFile -Encoding ASCII
```

As mentioned in https://github.com/lukesampson/scoop/pull/3212#issuecomment-472328841, we should encode file in "ASCII", since "UTF8" in Powershell 5- will be "UTF8-with-BOM", while "ASCII" be "UTF8".